samples/ipc_call_interface/producers/cipher_v2i1/cipher_v2i1.c

```diff
@@ -45,21 +147,62 @@ az_result cipher_v2i1_encrypt(
   AZ_ULIB_TRY
   {
     AZ_ULIB_THROW_IF_ERROR((context < NUMBER_OF_KEYS), AZ_ERROR_NOT_SUPPORTED);
-    AZ_ULIB_THROW_IF_ERROR((src_size + 1 <= dst_buffer_size), AZ_ERROR_NOT_ENOUGH_SPACE);
+    AZ_ULIB_THROW_IF_ERROR(
+        (encoded_len(src_size) + 1 <= dst_buffer_size), AZ_ERROR_NOT_ENOUGH_SPACE);
 
     uint32_t key_pos = 0;
 
     dst[0] = context + '0';
 
-    for (uint32_t i = 0; i < src_size; i++)
+    uint32_t destinationPosition = 1;
+    uint32_t currentPosition = 0;
+    char src_char[3];
+    while (src_size - currentPosition >= 3)
+    {
+      src_char[0] = src[currentPosition] ^ key[context][key_pos];
+      key_pos = next_key_pos(key_pos);
+      src_char[1] = src[currentPosition + 1] ^ key[context][key_pos];
+      key_pos = next_key_pos(key_pos);
+      src_char[2] = src[currentPosition + 2] ^ key[context][key_pos];
+      key_pos = next_key_pos(key_pos);
+
+      char c1 = base64char(src_char[0] >> 2);
```

Comment:
Is there currently a way to check this against DTDL?

Reply:
No, we do not have DTDL for this cipher interface.

---
In reply to: [582190403](https://github.com/Azure/azure-ulib-c/pull/60#discussion_r582190403) [](ancestors = 582190403)